packages/js/product-editor/src/components/variations-table/use-variations/use-variations.ts

```diff
@@ -432,16 +442,18 @@ export function useVariations( { productId }: UseVariationsProps ) {
 
 			currentPage++;
 
+			const deletedVariations = response?.delete ?? [];
 			result.push( ...( response?.delete ?? [] ) );
 
-			for ( const variation of subset ) {
+			for ( const variation of deletedVariations ) {
 				await coreInvalidateResolution( 'getEntityRecord', [
 					'postType',
 					'product_variation',
 					variation.id,
 				] );
 
-				onSelect( variation as never )( false );
+				delete selectedVariationsRef.current[ variation.id ];
+				setSelectedCount( ( current ) => current - 1 );
```

Comment:
Similar comment to the above, given `setSelectedCount` changes a state, how will this affect rendering when we do it in a for loop? Or will they all get combined into a single render?

Reply:
Yep, something like that. It is called [React state batch update](https://react.dev/learn/queueing-a-series-of-state-updates)